osgp/protocol-adapter-iec60870/osgp-protocol-adapter-iec60870/src/integration-test/java/org/opensmartgridplatform/adapter/protocol/iec60870/integrationtests/steps/MockedDeviceSteps.java

```diff
@@ -0,0 +1,107 @@
+package org.opensmartgridplatform.adapter.protocol.iec60870.integrationtests.steps;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.mockito.invocation.InvocationOnMock;
+import org.openmuc.j60870.ASdu;
+import org.openmuc.j60870.ASduType;
+import org.openmuc.j60870.CauseOfTransmission;
+import org.openmuc.j60870.Connection;
+import org.openmuc.j60870.ConnectionEventListener;
+import org.openmuc.j60870.ie.IeQualifierOfInterrogation;
+import org.openmuc.j60870.ie.IeSinglePointWithQuality;
+import org.openmuc.j60870.ie.InformationObject;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import io.cucumber.datatable.DataTable;
+import io.cucumber.java.en.Given;
+
+public class MockedDeviceSteps {
+    @Autowired
+    private ConnectionSteps connectionSteps;
+
+    @Autowired
+    private Connection connection;
+
+    @Given("a process image on the controlled station")
+    public void givenAProcessImageOnTheControlledStation(final DataTable datatable) throws Throwable {
```

Comment:
The Process Image should represent the actual status of all objects for a device. I agree with you that it would be convenient to have this class accessible broader than here. It could probably be moved to the osgp-iec60870 component.

I am not sure though about splitting up the method, at least not in this context.
In this case it is intended to mock a device to return a specific set of values when receiving a GI request. If splitting up, in my opinion it would be to another given step: givenTheDeviceRespondsToGeneralInterrogationWith or something like that, as I think the when step should be a trigger on the system/component under test (in this case the protocol adapter)?